.github/workflows/main.yml

```diff
@@ -55,32 +56,44 @@ jobs:
       - name: install requirements
         run: pip install -r requirements.txt
       - name: download ubuntu results
-        uses: actions/download-artifact@v1
+        uses: actions/download-artifact@v2
         with:
           name: results-ubuntu-latest
+          path: /tmp/results-ubuntu
       - name: download macos results
-        uses: actions/download-artifact@v1
+        uses: actions/download-artifact@v2
         with:
           name: results-macos-latest
+          path: /tmp/results-macos
       - name: download windows results
-        uses: actions/download-artifact@v1
+        uses: actions/download-artifact@v2
         with:
           name: results-windows-latest
+          path: /tmp/results-windows
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - name: dvc pull existing
+        run: dvc pull
       - name: join results
         run: |
-          mkdir results
-          cp -r results-macos-latest/* results/
-          cp -r results-windows-latest/* results/
-          cp -r results-ubuntu-latest/* results/
+          cp -r /tmp/results-windows/* results/
+          cp -r /tmp/results-macos/* results/
+          cp -r /tmp/results-ubuntu/* results/
+      - name: commit results to dvc
+        run: dvc commit -f run_benchmarks.dvc && dvc push
```

Comment:
Whoa, this is very neat! :fire: 

Have you looked into run-cache, btw? Seems like we could've made our dvc cache an artifact and then just `dvc pull --run-cache && dvc repro` here instead. Just a question, I really like your current approach too. :ok_hand:

Reply:
Though, with run-cache you won't even have to create a commit/PR. Unless I'm missing something here.

Reply:
I did not, will look into that soon.

Reply:
is there a reason for using dvc rather than the builtin action cache mechanism?

Reply:
@casperdcl Being able to pull(and even possibly push) results locally on our own machines.